services/console-proxy/server/src/main/java/com/cloud/consoleproxy/ConsoleProxyNoVncClient.java

```diff
@@ -113,32 +114,36 @@ public void run() {
                             if (client.isVncOverWebSocketConnectionOpen()) {
                                 updateFrontEndActivityTime();
                             }
-                            connectionAlive = client.isVncOverWebSocketConnectionAlive();
-                            try {
-                                Thread.sleep(1);
-                            } catch (Exception e) {
-                                s_logger.warn("Error on sleep for vnc over websocket", e);
-                            }
+                            connectionAlive = session.isOpen();
+                            Thread.sleep(1);
                         } else if (client.isVncOverNioSocket()) {
                             byte[] bytesArr;
                             int nextBytes = client.getNextBytes();
                             bytesArr = new byte[nextBytes];
                             client.readBytes(bytesArr, nextBytes);
+                            s_logger.trace(String.format("Read [%s] bytes from client [%s]", nextBytes, clientId));
                             if (nextBytes > 0) {
                                 session.getRemote().sendBytes(ByteBuffer.wrap(bytesArr));
                                 updateFrontEndActivityTime();
+                            } else {
+                                connectionAlive = session.isOpen();
                             }
                         } else {
                             b = new byte[100];
                             readBytes = client.read(b);
+                            s_logger.trace(String.format("Read [%s] bytes from client [%s]", readBytes, clientId));
                             if (readBytes == -1 || (readBytes > 0 && !sendReadBytesToNoVNC(b, readBytes))) {
                                 connectionAlive = false;
                             }
                         }
                     }
-                    connectionAlive = false;
+                    s_logger.info(String.format("Connection with client [%s] is dead.", clientId));
                 } catch (IOException e) {
                     s_logger.error("Error on VNC client", e);
+                } catch (InterruptedException e) {
```

Comment:
@JoaoJandre 
maybe better to keep the origin code and do not rhrow exception in case of InterruptedException

Reply:
@weizhouapache catching all exceptions is a bad practice, there's no reason to use that in this situation. But I can remove the exception being thrown and just keep the log.

Reply:
@JoaoJandre 
I agree if you change Exception to InterruptedException
I think InterruptedException should not terminate the process, it should be silently ignored as other processes.

Reply:
@JoaoJandre 
to be clear, I think we should use
```
                            try {
                                Thread.sleep(1);
                            } catch (InterruptedException e) {
                                s_logger.warn("Error on sleep for vnc over websocket", e);
                            }
```

Reply:
@JoaoJandre 
what's your opinion ?

Reply:
@weizhouapache I think it's fine either way, there are pros and cons, but I think that not terminating the process is a better option.